pkg/discovery/dns/provider.go

```diff
@@ -23,7 +24,7 @@ type Provider struct {
 	resolved map[string][]string
 	logger   log.Logger
 
-	resolverAddrs         *prometheus.GaugeVec
+	resolverAddrs         *extprom.TxGaugeVec
```

Comment:
Is this PR building on some PR that has not yet been merged? Or why is this change in here, when "just" updating the CI? :thinking:

Reply:
What do you mean? We need that for metric to be "atomic"